src/ServerNotifications/ServerNotification.php

```diff
@@ -65,32 +69,48 @@ class ServerNotification
      */
     protected $notificationType;
 
+    /**
+     * @param string $notificationType
+     */
+    public function __construct(string $notificationType)
+    {
+        $this->notificationType = $notificationType;
+    }
+
     /**
      * @param array $attributes
-     * @return static
+     * @return ServerNotification
      */
-    public static function fromArray(array $attributes): self
+    public static function fromArray(array $attributes = []): self
     {
-        $obj = new self();
```

Comment:
personally I'd also go fail-fast here to avoid side-effects instead of dealing with bunch of null values

Reply:
It's intended to make all the attributes optional!